tests/app/service/test_api_key_endpoints.py

```diff
@@ -49,6 +57,50 @@ def test_api_key_should_return_error_when_service_does_not_exist(notify_api):
             assert response.status_code == 404
 
 
+def test_api_key_should_return_error_when_user_does_not_exist(notify_api, sample_service):
+    service = sample_service()
+
+    with notify_api.test_request_context():
+        with notify_api.test_client() as client:
+            missing_user_id = str(uuid4())
+            data = {
+                'name': 'some secret name',
+                'created_by': missing_user_id,
+                'key_type': KEY_TYPE_NORMAL,
+            }
+            auth_header = create_admin_authorization_header()
+            response = client.post(
+                url_for('service.create_api_key', service_id=service.id),
+                data=json.dumps(data),
+                headers=[('Content-Type', 'application/json'), auth_header],
+            )
+            assert response.status_code == 400
+            assert 'IntegrityError' in response.json['message']
+
+
+def test_api_key_should_return_error_when_key_type_invlid(
+    notify_api,
+    sample_service,
+):
+    service = sample_service()
+
+    with notify_api.test_request_context():
+        with notify_api.test_client() as client:
+            data = {
+                'name': 'some secret name',
+                'created_by': str(uuid4()),
```

Comment:
How are we sure that the error isn't because this is a missing user?  This is exactly what you did in the previous test.  I would expect to have to create a user and explicitly use that user's ID.

Reply:
Good call. I mean to put the proper ID in that one. I was messing around with different ways to test it.